infrastructure/code/types.go

```diff
@@ -59,3 +59,46 @@ type MarkerPosition struct {
 	Rows CodePoint `json:"rows"`
 	File string    `json:"file"`
 }
+
+// AutofixResponse is the json-based structure to which we can translate the results of the HTTP
+// request to Autofix upstream.
+type AutofixResponse struct {
+	Status             string                      `json:"status"`
+	AutofixSuggestions []autofixResponseSuggestion `json:"fixes"`
+}
+
+type autofixResponseSuggestion = string
+
+type AutofixRequestKey struct {
+	Type     string `json:"type"`
+	Hash     string `json:"hash"`
+	Shard    string `json:"shard"`
+	FilePath string `json:"filePath"`
+	RuleId   string `json:"ruleId"`
+	LineNum  int    `json:"lineNum"` // 1-based
```

Comment:
Isn't it a problem/confusing that it's 1-based on backend, if e.g. analysis results are 0-based on the Code Analysis backend?

Reply:
I see your point, actually I was hesitating back and forth if to implement your comment. I confirm that the VScode and LSP are using 0-based ones. I double checked this
> analysis results are 0-based on the Code Analysis backend
and found that it is still 1-based in our API https://github.com/snyk/snyk-ls/blob/7667009db2846f270ab1b4fac2278133fc408836/infrastructure/code/convert.go#L283

This seems to follow the SARIF standard (https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html#_Ref493492556) which says
> The line number of the first line in a text artifact SHALL be 1

So here I vote to keep conistent with the backend API, I'll add links to comments and make convertation more prominent.

Reply:
✅

Reply:
It's a mess: two standards with different bases. Sigh.

Reply:
Nice find, @agronskiy 👍